src/nanoarrow/nanoarrow_device_cuda.c

```diff
@@ -246,8 +252,8 @@ static ArrowErrorCode ArrowDeviceCudaBufferCopyInternal(struct ArrowDevice* devi
     (*n_pop_context)++;
 
     NANOARROW_CUDA_RETURN_NOT_OK(
-        cuMemcpyDtoD((CUdeviceptr)dst.data.data, (CUdeviceptr)src.data.data,
-                     (size_t)src.size_bytes),
+        cuMemcpyDtoDAsync((CUdeviceptr)dst.data.data, (CUdeviceptr)src.data.data,
+                          (size_t)src.size_bytes, hstream),
         "cuMemcpytoD", error);
```

Comment:
May want to update to make this say `Async` as well?
```suggestion
        "cuMemcpyDtoD", error);
```